src/Bridge/Laravel/Http/Middlewares/AuditMiddleware.php

```diff
@@ -0,0 +1,166 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\Auditing\Bridge\Laravel\Http\Middlewares;
+
+use Closure;
+use DateTime as BaseDateTime;
+use EoneoPay\Externals\Logger\Interfaces\LoggerInterface;
+use EoneoPay\Utils\DateTime;
+use Exception;
+use Illuminate\Http\Request;
+use LoyaltyCorp\Auditing\Bridge\Laravel\Services\Interfaces\HttpLoggerInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+class AuditMiddleware
+{
+    /**
+     * @var \LoyaltyCorp\Auditing\Bridge\Laravel\Services\Interfaces\HttpLoggerInterface
+     */
+    private $httpLogger;
+
+    /**
+     * @var \EoneoPay\Externals\Logger\Interfaces\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var \Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface
+     */
+    private $psr7Factory;
+
+    /**
+     * AuditMiddleware constructor.
+     *
+     * @param \LoyaltyCorp\Auditing\Bridge\Laravel\Services\Interfaces\HttpLoggerInterface $httpLogger
+     * @param \EoneoPay\Externals\Logger\Interfaces\LoggerInterface $logger
+     * @param \Symfony\Bridge\PsrHttpMessage\HttpMessageFactoryInterface $psr7Factory
+     */
+    public function __construct(
+        HttpLoggerInterface $httpLogger,
+        LoggerInterface $logger,
+        HttpMessageFactoryInterface $psr7Factory
+    ) {
+        $this->httpLogger = $httpLogger;
+        $this->logger = $logger;
+        $this->psr7Factory = $psr7Factory;
+    }
+
+    /**
+     * Handle the request and response and convert them to psr7 to be logged
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure $next
+     *
+     * @return mixed
+     *
+     * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeStringException
+     * @throws \Exception
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        $datetime = new DateTime();
+        $originalException = null;
+        $response = null;
+
+        try {
+            $response = $next($request);
+        } catch (Exception $exception) {
+            $originalException = $exception;
+        }
+
+        $psrRequest = $this->createPsr7Request($request);
+
+        $psrResponse = null;
+        if (($response instanceof Response) === true) {
+            $psrResponse = $this->createPsr7Response($response);
+        }
+
+        // if response is already an instance of psr7
+        if (($response instanceof ResponseInterface) === true) {
+            $psrResponse = $response;
+        }
+
+        $this->callHttpLogger($datetime, $request->ip() ?? '', $psrRequest, $psrResponse);
```

Comment:
Could you move lines 76 through 88 into a private method `processForLogging()` and just call it with the $request and nullable $response?